src/components/mobile-menu/mobile-menu.jsx

```diff
@@ -0,0 +1,153 @@
+  state = {
+    isMenuOpen: false,
+    drawerPosition: 0,
+  }
+
+  static propTypes = {
+    stickyOffset: PropTypes.number,
+  }
+
+  static defaultProps = {
+    stickyOffset: 75,
+  }
+
+  componentDidMount() {
+    document.body.style.overflow = 'visible'
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <SwipeEventEmitter
+          onRelease={this.onSwipeRelease}
+          onHorizontalMove={this.updateDrawerPosition}
+        />
+        <Flex justifyContent="space-between" alignItems="center">
+          {this.props.renderLogo()}
+          <MenuIcon onToggle={this.toggle} isOpened={this.state.isMenuOpen} />
+        </Flex>
+        <Portal>
+          <Spring
+            to={this.animationPose}
+            immediate={name =>
+              this.state.drawerPosition !== 0 && name === 'translation'
+            }>
+            {({ translation, opacity }) => (
+              <MobileSidebar
+                translation={translation}
+                transitioned={this.state.drawerPosition === 0}>
+                {this.isSticky ? (
+                  <Flex
+                    justifyContent="space-between"
+                    alignItems="center"
+                    m={'2rem'}
+                    style={{
+                      opacity,
+                      position: 'absolute',
+                      top: 0,
+                      left: 0,
+                      right: 0,
+                    }}>
+                    {this.props.renderLogo()}
+                    <MenuIcon
+                      onToggle={this.toggle}
+                      isOpened={this.state.isMenuOpen}
+                    />
+                  </Flex>
+                ) : null}
+                {this.props.children}
+              </MobileSidebar>
+            )}
+          </Spring>
+        </Portal>
+      </React.Fragment>
+    )
+  }
+  get animationPose() {
+    const { isMenuOpen, drawerPosition } = this.state
+    let windowWidth = window.innerWidth
+    let finalDistance = ''
+    if (!isMenuOpen) {
+      finalDistance = drawerPosition
+    }
+    if (isMenuOpen) {
+      finalDistance = -windowWidth
+    }
+    if (isMenuOpen && drawerPosition > 0) {
+      finalDistance = -windowWidth + drawerPosition
+    }
+    return {
+      translation: finalDistance,
+      opacity: -finalDistance / windowWidth,
+    }
+  }
+  get isSticky() {
+    return this.props.stickyOffset < window.pageYOffset
+  }
+  updateDrawerPosition = event => {
+    this.setState(pState => ({ ...pState, drawerPosition: event.distance }))
+  }
+  onSwipeRelease = event => {
+    const { distance } = event
+    this.setState(pState => ({
+      ...pState,
+      drawerPosition: 0,
+    }))
+    if (distance < -130) {
+      return this.showMenu()
+    }
+    if (distance > 130) {
+      return this.hideMenu()
+    }
+  }
+  showMenu = () => {
+    document.body.style.overflow = 'hidden'
+    return this.setState(pState => ({
```

Comment:
I guess here you don't need to pass callback to `setState` - new stead doesn't depend in previous one

Reply:
I prefer to always use functional setState, I don't see any restrictions and limitations that we have with functional setState, and more on that year ago there had been rumors that object setState will be deprecated. But for the sake of consistency - fixed.

Reply:
I guess these 2 signatures can still be used for "communication" (like `let` and `const`)

if you want to say that you do not need previous state you pass object, otherwise - function

here I'm disappointed, cause you use function but you do not need previous state, and moreover you use spread though react performs shallow merge for you